### ISSUE TITLE
How do I include normal nifty xml files (into my main xml)?

### ISSUE THREAD

**Issue by spaceemotion** [nifty-gui/nifty-gui#259]
In my project setup, I have one `main.xml` that only registers bundles, effects, styles and controls.

I would like to include the xml files for my other screens as well though. Embedding styles / controls is fairly easy using the `useXXX` tag, but how do I do this with default xml files, that are not defining styles or controls?

I really would like to be able to update the files and includes from the files, and not the code. In code, I am currently calling Nifty's `addXml()` function for every screen xml that I have (not very dynamic).

Is this even possible / implemented?

**Comment by void256** [comment]
I don't get it :) You want to do what? o_O

You want to include xml files into other xml files? Maybe xinclude helps? http://cs.au.dk/~amoeller/XML/common/xinclude.html  I haven't tried it though :)

**Comment by spaceemotion** [comment]
Yes, that's basically what I want. However, I tried that and it doesn't seem to work…

**Comment by void256** [comment]
yes, just tried it and I think XPP3 does not support that :/

**Comment by spaceemotion** [comment]
Soo, I would basically have to tell my game to load a bunch of xmls based on a text file then? :/

**Comment by relu91** [comment]
I couldn't find anything about xpp3 and include support :( it's a pity that it could probably be not supported. Anyway if I was in you , I'll use a fixed path to load all the screens. For instance all the screens are in the Screens folder in your assets. It's convention over configuration.
